src/wallet/mod.rs

```diff
@@ -1581,15 +1581,37 @@ where
 /// Trait implemented by types that can be used to measure weight units.
 pub trait Vbytes {
     /// Convert weight units to virtual bytes.
-    fn vbytes(self) -> f32;
+    fn vbytes(self) -> usize;
 }
 
 impl Vbytes for usize {
-    fn vbytes(self) -> f32 {
-        self as f32 / 4.0
+    fn vbytes(self) -> usize {
+        // ref: https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki#transaction-size-calculations
+        (self as f32 / 4.0).ceil() as usize
     }
 }
 
+/// Calculate fee rate by weight units.
+pub fn fee_rate_wu(fee: u64, wu: usize) -> FeeRate {
+    fee_rate_vbytes(fee, wu.vbytes())
+}
+
+/// Calculate fee rate by virtual bytes.
+pub fn fee_rate_vbytes(fee: u64, vbytes: usize) -> FeeRate {
+    let rate = fee as f32 / vbytes as f32;
+    FeeRate::from_sat_per_vb(rate)
+}
+
+/// Calculate absolute fee in Satoshis from fee rate and weight unit size.
+pub fn fee_wu(fee_rate: FeeRate, wu: usize) -> u64 {
+    fee_vbytes(fee_rate, wu.vbytes())
+}
+
+/// Calculate absolute fee in Satoshis from fee rate and virtual byte size.
+pub fn fee_vbytes(fee_rate: FeeRate, vbytes: usize) -> u64 {
+    (fee_rate.as_sat_vb() * vbytes as f32).ceil() as u64
+}
+
```

Comment:
Is there a reason why these functions aren't constructors implemented on `FeeRate` directly?

Reply:
None at all now you mention it. Will play around with this idea.

Reply:
Much nicer, thanks! I shortened the names also in line with the other `FeeRate` methods.